diff --git a/drivers/mtd/spi-nor/gigadevice.c b/drivers/mtd/spi-nor/gigadevice.c
index 252c8d735b25..329ddfe8bf3d 100644
--- a/drivers/mtd/spi-nor/gigadevice.c
+++ b/drivers/mtd/spi-nor/gigadevice.c
@@ -22,6 +22,14 @@ static void gd25q256_post_sfdp(struct spi_nor *nor)
 static struct spi_nor_fixups gd25q256_fixups = {
 	.post_sfdp  = gd25q256_post_sfdp,
 };
+static void gd25lb512me_default_init(struct spi_nor *nor)
+{
+	nor->params->quad_enable = spi_nor_sr1_bit6_quad_enable;
+}
+
+static const struct spi_nor_fixups gd25lb512me_fixups = {
+	.default_init = gd25lb512me_default_init,
+};

 static const struct flash_info gigadevice_parts[] = {
 	{ "gd25q16", INFO(0xc84015, 0, 64 * 1024,  32,
@@ -66,6 +74,11 @@ static const struct flash_info gigadevice_parts[] = {
 			   SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ |
 			   SPI_NOR_4B_OPCODES | SPI_NOR_HAS_LOCK |
 			   SPI_NOR_HAS_TB) },
+	{ "gd25lb512me", INFO(0xc8671a, 0, 64 * 1024, 1024,
+		SPI_NOR_HAS_LOCK | SPI_NOR_HAS_TB |
+		SECT_4K /* | SPI_NOR_QUAD_READ */       |
+		SPI_NOR_4B_OPCODES)
+		.fixups = &gd25lb512me_fixups },
 };

 const struct spi_nor_manufacturer spi_nor_gigadevice = {
