diff -Naur '--exclude=.git' a/Makefile.am b/Makefile.am
--- a/Makefile.am	2020-08-11 11:18:29.132208483 +0800
+++ b/Makefile.am	2020-08-11 11:46:48.424313308 +0800
@@ -12,7 +12,7 @@
 		xyz.openbmc_project.Network.conf
 endif
 
-bin_PROGRAMS = phosphor-network-manager ncsi-netlink
+bin_PROGRAMS = phosphor-network-manager
 
 noinst_HEADERS = \
 		ethernet_interface.hpp \
@@ -33,19 +33,6 @@
 		watch.hpp \
 		argument.hpp
 
-ncsi_netlink_SOURCES = \
-		argument.cpp \
-		ncsi_netlink_main.cpp \
-		ncsi_util.cpp
-
-ncsi_netlink_LDFLAGS = \
-        $(PHOSPHOR_LOGGING_LIBS) \
-		$(LIBNL_LIBS)
-
-ncsi_netlink_CPPFLAGS = $(LIBNL_CFLAGS)
-ncsi_netlink_CXXFLAGS = -flto
-
-
 phosphor_network_manager_SOURCES = \
 		ethernet_interface.cpp \
 		neighbor.cpp \

