From b720e4766683f6447ca054e67fd4aebe175e8e5e Mon Sep 17 00:00:00 2001
From: Troy Lee <troy_lee@aspeedtech.com>
Date: Thu, 25 May 2023 14:14:41 +0800
Subject: [PATCH] bmcweb: fixes virtual media buffer overflow

Signed-off-by: Troy Lee <troy_lee@aspeedtech.com>
---
 include/vm_websocket.hpp | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/include/vm_websocket.hpp b/include/vm_websocket.hpp
index 284b5d2e..5723164e 100644
--- a/include/vm_websocket.hpp
+++ b/include/vm_websocket.hpp
@@ -63,7 +63,15 @@ class Handler : public std::enable_shared_from_this<Handler>
             return;
         }
 
-        proxy.wait();
+        boost::system::error_code ec;
+        do
+        {
+            proxy.wait(ec);
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("Error on proxy.wait(): {}", ec.message());
+            }
+        } while (ec == boost::system::errc::interrupted);
     }
 
     void connect()
