From 60b1e005586597deeddc285af8829375736be875 Mon Sep 17 00:00:00 2001
From: VinceChang6637 <vince_chang@aspeedtech.com>
Date: Fri, 7 Jun 2024 13:51:44 +0800
Subject: [PATCH 2/3] fansensor support ast2700 pwm driver

Signed-off-by: VinceChang6637 <vince_chang@aspeedtech.com>
---
 src/FanMain.cpp | 2 ++
 1 file changed, 2 insertions(+)

Index: git/src/FanMain.cpp
===================================================================
--- git.orig/src/FanMain.cpp
+++ git/src/FanMain.cpp
@@ -81,6 +81,7 @@ static const std::map<std::string, FanTy
     {"aspeed,ast2400-pwm-tacho", FanTypes::aspeed},
     {"aspeed,ast2500-pwm-tacho", FanTypes::aspeed},
     {"aspeed,ast2600-pwm-tach", FanTypes::aspeed},
+    {"aspeed,ast2700-pwm-tach", FanTypes::aspeed},
     {"nuvoton,npcm750-pwm-fan", FanTypes::nuvoton},
     {"nuvoton,npcm845-pwm-fan", FanTypes::nuvoton},
     {"hpe,gxp-fan-ctrl", FanTypes::hpe}
