From d78868ec616c91ead0263f3891b23c86c097e881 Mon Sep 17 00:00:00 2001
From: Jamin Lin <jamin_lin@aspeedtech.com>
Date: Wed, 16 Nov 2022 13:45:06 +0800
Subject: [PATCH] Add tx retry mechanism in transmitQueuedMessages

Signed-off-by: Jamin Lin <jamin_lin@aspeedtech.com>
---
 src/utils/transmission_queue.cpp | 25 +++++++++++++++++++------
 1 file changed, 19 insertions(+), 6 deletions(-)

diff --git a/src/utils/transmission_queue.cpp b/src/utils/transmission_queue.cpp
index 859cbc8..20e89ca 100644
--- a/src/utils/transmission_queue.cpp
+++ b/src/utils/transmission_queue.cpp
@@ -61,13 +61,26 @@ void MctpTransmissionQueue::Endpoint::transmitQueuedMessages(struct mctp* mctp,
         auto message = std::move(queuedMessageIter->second);
         queuedMessages.erase(queuedMessageIter);
 
-        int rc = mctp_message_tx(mctp, destEid, message->payload.data(),
-                                 message->payload.size(), true, msgTag,
-                                 message->privateData.data());
-        if (rc < 0)
+        int retry = 5;
+        int rc;
+        int i;
+
+        for (i = 0; i <= retry; i++) {
+            rc = mctp_message_tx(mctp, destEid, message->payload.data(),
+                                    message->payload.size(), true, msgTag,
+                                    message->privateData.data());
+            if (rc == 0)
+                break;
+             phosphor::logging::log<phosphor::logging::level::ERR>(
+                ("Error in mctp_message_tx: retry..."+ std::to_string(i)).c_str());
+             usleep(10*1000);
+        }
+
+        //if (rc < 0)
+        if (i > retry)
         {
             phosphor::logging::log<phosphor::logging::level::ERR>(
-                "Error in mctp_message_tx");
+                ("Error in mctp_message_tx"+ std::to_string(rc)).c_str());
             continue;
         }
 
@@ -129,4 +142,4 @@ void MctpTransmissionQueue::dispose(mctp_eid_t destEid,
             endpoint.transmittedMessages.erase(transmittedMessageIter);
         }
     }
-}
\ No newline at end of file
+}
-- 
2.25.1

