From 190134809587f8e99f421b6d13ca8fda1bafa653 Mon Sep 17 00:00:00 2001
From: Jamin Lin <jamin_lin@aspeedtech.com>
Date: Tue, 19 Jul 2022 17:19:57 +0800
Subject: [PATCH] disable all warnings being treated as errors

Signed-off-by: Jamin Lin <jamin_lin@aspeedtech.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7ad941c..0b579a5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,7 +13,6 @@ add_definitions(-DMCTP_ASTPCIE_RESPONSE_WA)
 
 set(CMAKE_CXX_FLAGS
     "${CMAKE_CXX_FLAGS} \
-    -Werror \
     -Wall \
     -Wextra \
     -Wshadow \
-- 
2.25.1

