From e67ac7ce930adb0a5840cda2adc56be5ebcae372 Mon Sep 17 00:00:00 2001
From: Troy Lee <troy_lee@aspeedtech.com>
Date: Wed, 20 Jul 2022 09:45:35 +0800
Subject: [PATCH] bmcweb: fixes virtual media buffer overflow

---
 include/vm_websocket.hpp | 20 ++++++++++++++++++--
 1 file changed, 18 insertions(+), 2 deletions(-)

diff --git a/include/vm_websocket.hpp b/include/vm_websocket.hpp
index 188bcf2ad..5cfc6a83a 100644
--- a/include/vm_websocket.hpp
+++ b/include/vm_websocket.hpp
@@ -48,7 +48,16 @@ class Handler : public std::enable_shared_from_this<Handler>
         {
             return;
         }
-        proxy.wait();
+
+        std::error_code ec;
+        do
+        {
+            proxy.wait(ec);
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "Error on proxy.wait():" << ec;
+            }
+        } while (ec == std::errc::interrupted);
     }
 
     void connect()
@@ -205,8 +214,15 @@ inline void requestRoutes(App& app)
         })
         .onmessage([](crow::websocket::Connection& conn,
                       const std::string& data, bool) {
+            if (handler->inputBuffer->size() > 0)
+            {
+                BMCWEB_LOG_DEBUG << "Data length " << data.length()
+                                 << " iBuffer.cap() " << handler->inputBuffer->capacity()
+                                 << " iBuffer.size() " << handler->inputBuffer->size();
+            }
+
             if (data.length() >
-                handler->inputBuffer->capacity() - handler->inputBuffer->size())
+                    handler->inputBuffer->capacity() - handler->inputBuffer->size())
             {
                 BMCWEB_LOG_ERROR << "Buffer overrun when writing "
                                  << data.length() << " bytes";
-- 
2.25.1

