diff --git a/redfish-core/lib/systems.hpp b/redfish-core/lib/systems.hpp
index 4dc75409..71e9179a 100644
--- a/redfish-core/lib/systems.hpp
+++ b/redfish-core/lib/systems.hpp
@@ -3463,4 +3463,38 @@ inline void requestRoutesSystemResetActionInfo(App& app)
         asyncResp->res.jsonValue["Parameters"] = std::move(parameters);
         });
 }
+
+
+void handleSolLogsGet(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp) {
+    std::string filePath = "/var/log/obmc-console.log"; // 日志文件路径
+    std::ifstream logFile(filePath, std::ios::in | std::ios::binary);
+
+    if (logFile) {
+        std::ostringstream contents;
+        contents << logFile.rdbuf(); // 读取文件内容
+        logFile.close();
+
+        asyncResp->res.addHeader("Content-Type", "text/plain");
+        asyncResp->res.body() = contents.str();
+    } else {
+        // 文件打开失败，返回错误
+        messages::resourceNotFound(asyncResp->res, "Log file", filePath);
+    }
+}
+
+inline void requestRoutesSystemsSol(App& app) {
+    BMCWEB_ROUTE(app, "/sol/logs")
+        .privileges({}) // TODO: Set the correct privileges if needed
+        .methods(boost::beast::http::verb::get)(
+            [&app](const crow::Request& req,
+                   const std::shared_ptr<bmcweb::AsyncResp>& asyncResp) {
+                if (!redfish::setUpRedfishRoute(app, req, asyncResp)) {
+                    BMCWEB_LOG_ERROR << "setUpRedfishRoute error";
+                    return;
+                }
+                handleSolLogsGet(asyncResp);
+            });
+}
+
+
 } // namespace redfish
