From b3af668361628f9e1049374b04b6ef4d75c976b1 Mon Sep 17 00:00:00 2001
From: zhaohui-yan <zhaohui.yan@sophgo.com>
Date: Wed, 14 Aug 2024 17:45:26 +0800
Subject: [PATCH 1/1] update inventory UI.

---
 .../HardwareStatus/Inventory/Inventory.vue    | 80 +++++++++----------
 1 file changed, 40 insertions(+), 40 deletions(-)

diff --git a/src/views/HardwareStatus/Inventory/Inventory.vue b/src/views/HardwareStatus/Inventory/Inventory.vue
index fcdbf8d..7030236 100644
--- a/src/views/HardwareStatus/Inventory/Inventory.vue
+++ b/src/views/HardwareStatus/Inventory/Inventory.vue
@@ -52,13 +52,13 @@
 import PageTitle from '@/components/Global/PageTitle';
 import ServiceIndicator from './InventoryServiceIndicator';
 import TableSystem from './InventoryTableSystem';
-import TablePowerSupplies from './InventoryTablePowerSupplies';
-import TableDimmSlot from './InventoryTableDimmSlot';
-import TableFans from './InventoryTableFans';
+// import TablePowerSupplies from './InventoryTablePowerSupplies';
+// import TableDimmSlot from './InventoryTableDimmSlot';
+// import TableFans from './InventoryTableFans';
 import TableBmcManager from './InventoryTableBmcManager';
 import TableChassis from './InventoryTableChassis';
-import TableProcessors from './InventoryTableProcessors';
-import TableAssembly from './InventoryTableAssembly';
+// import TableProcessors from './InventoryTableProcessors';
+// import TableAssembly from './InventoryTableAssembly';
 import LoadingBarMixin from '@/components/Mixins/LoadingBarMixin';
 import PageSection from '@/components/Global/PageSection';
 import JumpLink16 from '@carbon/icons-vue/es/jump-link/16';
@@ -69,14 +69,14 @@ export default {
   components: {
     PageTitle,
     ServiceIndicator,
-    TableDimmSlot,
-    TablePowerSupplies,
+    // TableDimmSlot,
+    // TablePowerSupplies,
     TableSystem,
-    TableFans,
+    // TableFans,
     TableBmcManager,
     TableChassis,
-    TableProcessors,
-    TableAssembly,
+    // TableProcessors,
+    // TableAssembly,
     PageSection,
     JumpLink: JumpLink16,
   },
@@ -108,36 +108,36 @@ export default {
           href: '#chassis',
           linkText: this.$t('pageInventory.chassis'),
         },
-        {
-          id: 'dimms',
-          dataRef: 'dimms',
-          href: '#dimms',
-          linkText: this.$t('pageInventory.dimmSlot'),
-        },
-        {
-          id: 'fans',
-          dataRef: 'fans',
-          href: '#fans',
-          linkText: this.$t('pageInventory.fans'),
-        },
-        {
-          id: 'powerSupply',
-          dataRef: 'powerSupply',
-          href: '#powerSupply',
-          linkText: this.$t('pageInventory.powerSupplies'),
-        },
-        {
-          id: 'processors',
-          dataRef: 'processors',
-          href: '#processors',
-          linkText: this.$t('pageInventory.processors'),
-        },
-        {
-          id: 'assembly',
-          dataRef: 'assembly',
-          href: '#assembly',
-          linkText: this.$t('pageInventory.assemblies'),
-        },
+        // {
+        //   id: 'dimms',
+        //   dataRef: 'dimms',
+        //   href: '#dimms',
+        //   linkText: this.$t('pageInventory.dimmSlot'),
+        // },
+        // {
+        //   id: 'fans',
+        //   dataRef: 'fans',
+        //   href: '#fans',
+        //   linkText: this.$t('pageInventory.fans'),
+        // },
+        // {
+        //   id: 'powerSupply',
+        //   dataRef: 'powerSupply',
+        //   href: '#powerSupply',
+        //   linkText: this.$t('pageInventory.powerSupplies'),
+        // },
+        // {
+        //   id: 'processors',
+        //   dataRef: 'processors',
+        //   href: '#processors',
+        //   linkText: this.$t('pageInventory.processors'),
+        // },
+        // {
+        //   id: 'assembly',
+        //   dataRef: 'assembly',
+        //   href: '#assembly',
+        //   linkText: this.$t('pageInventory.assemblies'),
+        // },
       ],
     };
   },
-- 
2.25.1

