From 5702ea5333fd77c15515f821283428a746b0370f Mon Sep 17 00:00:00 2001
From: zhaohui-yan <zhaohui.yan@sophgo.com>
Date: Wed, 14 Aug 2024 17:56:50 +0800
Subject: [PATCH 1/1] add SOL log export function.

---
 .../SerialOverLan/SerialOverLanConsole.vue    | 127 +++++++++++++++++-
 1 file changed, 126 insertions(+), 1 deletion(-)

diff --git a/src/views/Operations/SerialOverLan/SerialOverLanConsole.vue b/src/views/Operations/SerialOverLan/SerialOverLanConsole.vue
index 78b641c..50dbbf2 100644
--- a/src/views/Operations/SerialOverLan/SerialOverLanConsole.vue
+++ b/src/views/Operations/SerialOverLan/SerialOverLanConsole.vue
@@ -62,6 +62,28 @@
         </b-form-group>
       </b-form>
     </div>
+    <dt>
+          {{ $t('pageSerialOverLan.blank') }}
+    </dt>
+    <dt>
+          {{ $t('pageSerialOverLan.blank') }}
+    </dt>
+    <dt>
+          {{ $t('pageSerialOverLan.blank') }}
+    </dt>
+    <div>
+      <label for="logContent" class="bold-black-label">SOL log export:</label>
+    </div>
+    <div class="container">
+      <div class="button-container">
+        <button @click="fetchLogFile">Load</button>
+        <button @click="clearLogContent">Clear</button>
+        <button @click="downloadLogFile" :disabled="!logContent">Save</button>
+      </div>
+      <div class="textarea-container">
+        <textarea v-model="logContent" class="log-content" readonly></textarea>
+      </div>
+    </div>
   </div>
 </template>
 
@@ -73,7 +95,7 @@ import { Terminal } from 'xterm';
 import { throttle } from 'lodash';
 import IconLaunch from '@carbon/icons-vue/es/launch/20';
 import StatusIcon from '@/components/Global/StatusIcon';
-
+import axios from 'axios';
 export default {
   name: 'SerialOverLanConsole',
   components: {
@@ -97,6 +119,7 @@ export default {
       form: {
         port: '',
       },
+      logContent: '',
     };
   },
   computed: {
@@ -198,6 +221,31 @@ export default {
         this.$store.dispatch('controls/setSolToPort1');
       }
     },
+
+    async fetchLogFile() {
+      try {
+        const response = await axios.get('/sol/logs');
+        this.logContent = response.data;
+      } catch (error) {
+        console.error('Error fetching log file:', error);
+        this.logContent = 'Failed to load log file.';
+      }
+    },
+    downloadLogFile() {
+      const blob = new Blob([this.logContent], { type: 'text/plain' });
+      const url = URL.createObjectURL(blob);
+      const link = document.createElement('a');
+      link.href = url;
+      link.download = 'sol-log.txt';
+      document.body.appendChild(link);
+      link.click();
+      document.body.removeChild(link);
+      URL.revokeObjectURL(url);
+    },
+
+    clearLogContent() {
+      this.logContent = ''; // 清除logContent的内容
+    },
   },
 };
 </script>
@@ -214,3 +262,80 @@ export default {
   margin: 1.5%;
 }
 </style>
+
+<style scoped>
+
+.container {
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  padding-left: 0;
+  padding-right: 0;
+  height: 200px;
+  border: 2px solid #c8c2c2;
+  width: 100%;
+}
+
+.button-container {
+  display: flex;
+  flex-direction: column; /* 垂直排列按钮 */
+  justify-content: space-between;
+  padding-left: 0; /* 消除内边距 */
+  margin-left: 0px; /* 消除外边距 */
+  margin-right: 0px;
+  height: 100%;
+  background-color: #f2eeee;
+}
+
+.button-container button {
+  padding: 5px 16px;
+  margin-left: 0; /* 消除按钮左边距 */
+  color: #fff;
+  background-color: #2d60e5;
+  border-color: #d9dadc;
+  border-radius: 4px;
+  margin-left: 10px;
+  margin-right: 10px;
+
+}
+
+.button-container button:first-child {
+  margin-top: 10px;
+}
+
+.button-container button:last-child {
+  margin-bottom: 10px;
+}
+
+.textarea-container {
+  flex-grow: 1;
+  flex-direction: column;
+  justify-content: space-between;
+  flex-grow: 1;
+  width: 100%;
+  height: 100%;
+}
+.log-content {
+  width: 100%;
+  /* margin-top: 5px; */
+  box-sizing: border-box;
+  resize: none;
+  background-color: #19273c;
+  color: white;
+  height: 100%;
+}
+
+
+
+.bold-black-label {
+  margin-bottom: 5px;
+  /* font-weight: bold; */
+  color: #353333;
+  font-size: 20px;
+}
+
+
+.spacer {
+  flex-grow: 1;
+}
+</style>
\ No newline at end of file
-- 
2.25.1

