diff --git a/src/env/router/intel.js b/src/env/router/intel.js
index fd8ed77..a4c5240 100644
--- a/src/env/router/intel.js
+++ b/src/env/router/intel.js
@@ -171,6 +171,14 @@ const routes = [
           title: i18n.t('appPageTitle.dateTime'),
         },
       },
+      {
+        path: '/settings/fan-speed',
+        name: 'fan-speed',
+        component: FanSpeed,
+        meta: {
+          title: i18n.t('appPageTitle.fanSpeed'),
+        },
+      },
       {
         path: '/operations/kvm',
         name: 'kvm',
