diff --git a/src/store/modules/GlobalStore.js b/src/store/modules/GlobalStore.js
index 95d7a08..fa2d644 100644
--- a/src/store/modules/GlobalStore.js
+++ b/src/store/modules/GlobalStore.js
@@ -34,6 +34,12 @@ const GlobalStore = {
     modelType: null,
     serialNumber: null,
     serverStatus: 'unreachable',
+    cpuaPWStatus:'unreachable',
+    cpubPWStatus:'unreachable',
+    cpuoPWStatus:'unreachable',
+    otherPWStatus:'unreachable',
+    psuStatus:'unreachable',
+    efusStatus:'unreachable',
     languagePreference: localStorage.getItem('storedLanguage') || 'en-US',
     isUtcDisplay: localStorage.getItem('storedUtcDisplay')
       ? JSON.parse(localStorage.getItem('storedUtcDisplay'))
@@ -46,6 +52,12 @@ const GlobalStore = {
     modelType: (state) => state.modelType,
     serialNumber: (state) => state.serialNumber,
     serverStatus: (state) => state.serverStatus,
+    cpuaPWStatus:  (state)  => state.cpuaPWStatus,
+    cpubPWStatus:  (state)  => state.cpubPWStatus,
+    cpuoPWStatus:  (state)  => state.cpuoPWStatus,
+    otherPWStatus: (state)  => state.otherPWStatus,
+    psuStatus:     (state)  => state.psuStatus,
+    efusStatus:    (state)  => state.efusStatus,
     bmcTime: (state) => state.bmcTime,
     languagePreference: (state) => state.languagePreference,
     isUtcDisplay: (state) => state.isUtcDisplay,
@@ -60,6 +72,12 @@ const GlobalStore = {
     setBmcTime: (state, bmcTime) => (state.bmcTime = bmcTime),
     setServerStatus: (state, serverState) =>
       (state.serverStatus = serverStateMapper(serverState)),
+    setCpuaPWInfo: (state, cpuaPWInfo)  => (state.cpuaPWStatus  = cpuaPWInfo),
+    setCpubPWInfo: (state, cpubPWInfo)  => (state.cpubPWStatus  = cpubPWInfo),
+    setCpuoPWInfo: (state, cpuoPWInfo)  => (state.cpuoPWStatus  = cpuoPWInfo),
+    setOtherPWInfo:(state, otherPWInfo) => (state.otherPWStatus = otherPWInfo),
+    setPsuInfo:    (state, psuInfo)     => (state.psuStatus     = psuInfo),
+    setEfusInfo:   (state, efusInfo)    => (state.efusStatus    = efusInfo),
     setLanguagePreference: (state, language) =>
       (state.languagePreference = language),
     setUsername: (state, username) => (state.username = username),
@@ -91,6 +109,12 @@ const GlobalStore = {
               AssetTag,
               Model,
               PowerState,
+              CpuaPWInfo,
+              CpubPWInfo,
+              CpuoPWInfo,
+              OtherPWInfo,
+              PsuInfo,
+              EfusInfo,
               SerialNumber,
               Status: { State } = {},
             },
@@ -98,6 +122,12 @@ const GlobalStore = {
             commit('setAssetTag', AssetTag);
             commit('setSerialNumber', SerialNumber);
             commit('setModelType', Model);
+            commit('setCpuaPWInfo', CpuaPWInfo);
+            commit('setCpubPWInfo', CpubPWInfo);
+            commit('setCpuoPWInfo', CpuoPWInfo);
+            commit('setOtherPWInfo', OtherPWInfo);
+            commit('setPsuInfo', PsuInfo);
+            commit('setEfusInfo', EfusInfo);
             if (State === 'Quiesced' || State === 'InTest') {
               // OpenBMC's host state interface is mapped to 2 Redfish
               // properties "Status""State" and "PowerState". Look first
