diff --git a/src/store/modules/GlobalStore.js b/src/store/modules/GlobalStore.js
index bc3603d..0af86af 100644
--- a/src/store/modules/GlobalStore.js
+++ b/src/store/modules/GlobalStore.js
@@ -49,6 +49,7 @@ const GlobalStore = {
     fan7Status: null,
     fan8Status: null,
     fan9Status: null,
+    cpldVersion: null,
     languagePreference: localStorage.getItem('storedLanguage') || 'en-US',
     isUtcDisplay: localStorage.getItem('storedUtcDisplay')
       ? JSON.parse(localStorage.getItem('storedUtcDisplay'))
@@ -76,6 +77,7 @@ const GlobalStore = {
     fan7Status:    (state)  => state.fan7Status,
     fan8Status:    (state)  => state.fan8Status,
     fan9Status:    (state)  => state.fan9Status,
+    cpldVersion:   (state)  => state.cpldVersion,
     bmcTime: (state) => state.bmcTime,
     languagePreference: (state) => state.languagePreference,
     isUtcDisplay: (state) => state.isUtcDisplay,
@@ -105,6 +107,7 @@ const GlobalStore = {
     setFan7Info:  (state, fan7Info)   => (state.fan7Status    =  fan7Info),
     setFan8Info:  (state, fan8Info)   => (state.fan8Status    =  fan8Info),
     setFan9Info:  (state, fan9Info)   => (state.fan9Status    =  fan9Info),
+    setCpldInfo:  (state, cpldInfo)   => (state.cpldVersion   =  cpldInfo),
     setLanguagePreference: (state, language) =>
       (state.languagePreference = language),
     setUsername: (state, username) => (state.username = username),
@@ -151,6 +154,7 @@ const GlobalStore = {
               Fan7Info,
               Fan8Info,
               Fan9Info,
+              CpldInfo,
               SerialNumber,
               Status: { State } = {},
             },
@@ -173,6 +177,7 @@ const GlobalStore = {
             commit('setFan7Info',  Fan7Info );
             commit('setFan8Info',  Fan8Info );
             commit('setFan9Info',  Fan9Info );
+            commit('setCpldInfo',  CpldInfo );
             if (State === 'Quiesced' || State === 'InTest') {
               // OpenBMC's host state interface is mapped to 2 Redfish
               // properties "Status""State" and "PowerState". Look first
