diff --git a/src/env/components/AppNavigation/intel.js b/src/env/components/AppNavigation/intel.js
index 3fe0ad1..88108af 100644
--- a/src/env/components/AppNavigation/intel.js
+++ b/src/env/components/AppNavigation/intel.js
@@ -113,6 +113,11 @@ const AppNavigationMixin = {
               label: this.$t('appNavigation.network'),
               route: '/settings/network',
             },
+            {
+              id: 'fanspeed',
+              label: this.$t('appNavigation.fanSpeed'),
+              route: '/settings/fan-speed',
+            },
           ],
         },
         {
