diff --git a/src/env/router/intel.js b/src/env/router/intel.js
index fd8ed77..333a026 100644
--- a/src/env/router/intel.js
+++ b/src/env/router/intel.js
@@ -26,6 +26,7 @@ import Certificates from '@/views/SecurityAndAccess/Certificates';
 import VirtualMedia from '@/views/Operations/VirtualMedia';
 import Power from '@/views/ResourceManagement/Power';
 import i18n from '@/i18n';
+import FanSpeed from '@/views/Settings/FanSpeed';
 
 const routes = [
   {
@@ -171,6 +172,14 @@ const routes = [
           title: i18n.t('appPageTitle.dateTime'),
         },
       },
+      {
+        path: '/settings/fan-speed',
+        name: 'fan-speed',
+        component: FanSpeed,
+        meta: {
+          title: i18n.t('appPageTitle.fanSpeed'),
+        },
+      },
       {
         path: '/operations/kvm',
         name: 'kvm',
