diff --git a/src/router/routes.js b/src/router/routes.js
index 3cbdabc..596b6d4 100644
--- a/src/router/routes.js
+++ b/src/router/routes.js
@@ -30,6 +30,7 @@ import Certificates from '@/views/SecurityAndAccess/Certificates';
 import VirtualMedia from '@/views/Operations/VirtualMedia';
 import Power from '@/views/ResourceManagement/Power';
 import i18n from '@/i18n';
+import FanSpeed from '@/views/Settings/FanSpeed';
 
 const routes = [
   {
@@ -183,6 +184,14 @@ const routes = [
           title: i18n.t('appPageTitle.dateTime'),
         },
       },
+      {
+        path: '/settings/fan-speed',
+        name: 'fan-speed',
+        component: FanSpeed,
+        meta: {
+          title: i18n.t('appPageTitle.fanSpeed'),
+        },
+      },
       {
         path: '/operations/factory-reset',
         name: 'factory-reset',
