diff --git a/src/store/plugins/WebSocketPlugin.js b/src/store/plugins/WebSocketPlugin.js
index 7dd15f4..b5492a4 100644
--- a/src/store/plugins/WebSocketPlugin.js
+++ b/src/store/plugins/WebSocketPlugin.js
@@ -52,9 +52,9 @@ const WebSocketPlugin = (store) => {
         }
       } else if (path === '/xyz/openbmc_project/logging') {
         store.dispatch('eventLog/getEventLogData');
-      } else if ((path === '/xyz/openbmc_project/gpio/fan') ||
+      } else if (/* (path === '/xyz/openbmc_project/gpio/fan') ||
                  (path === '/xyz/openbmc_project/gpio/hdd') ||
-                 (path === '/xyz/openbmc_project/gpio/sata')||
+                 (path === '/xyz/openbmc_project/gpio/sata')|| */
                  (path === '/xyz/openbmc_project/gpio/identifyLed')) {
         store.dispatch('global/getSystemInfo');
       }
