diff --git a/src/store/modules/GlobalStore.js b/src/store/modules/GlobalStore.js
index 0af86af..077841c 100644
--- a/src/store/modules/GlobalStore.js
+++ b/src/store/modules/GlobalStore.js
@@ -50,6 +50,8 @@ const GlobalStore = {
     fan8Status: null,
     fan9Status: null,
     cpldVersion: null,
+    sg1684xStatus:null,
+    sg1684xTmp:null,
     languagePreference: localStorage.getItem('storedLanguage') || 'en-US',
     isUtcDisplay: localStorage.getItem('storedUtcDisplay')
       ? JSON.parse(localStorage.getItem('storedUtcDisplay'))
@@ -78,6 +80,8 @@ const GlobalStore = {
     fan8Status:    (state)  => state.fan8Status,
     fan9Status:    (state)  => state.fan9Status,
     cpldVersion:   (state)  => state.cpldVersion,
+    sg1684xStatus: (state)  => state.sg1684xStatus,
+    sg1684xTmp:    (state)  => state.sg1684xTmp,
     bmcTime: (state) => state.bmcTime,
     languagePreference: (state) => state.languagePreference,
     isUtcDisplay: (state) => state.isUtcDisplay,
@@ -108,6 +112,8 @@ const GlobalStore = {
     setFan8Info:  (state, fan8Info)   => (state.fan8Status    =  fan8Info),
     setFan9Info:  (state, fan9Info)   => (state.fan9Status    =  fan9Info),
     setCpldInfo:  (state, cpldInfo)   => (state.cpldVersion   =  cpldInfo),
+    setsg1684xInfo: (state, sg1684xInfo)   => (state.sg1684xStatus   =  sg1684xInfo),
+    setsg1684xTmpInfo: (state, sg1684xTmpInfo)   => (state.sg1684xTmp   =  sg1684xTmpInfo),
     setLanguagePreference: (state, language) =>
       (state.languagePreference = language),
     setUsername: (state, username) => (state.username = username),
@@ -155,6 +161,8 @@ const GlobalStore = {
               Fan8Info,
               Fan9Info,
               CpldInfo,
+              sg1684xInfo,
+              sg1684xTmpInfo,
               SerialNumber,
               Status: { State } = {},
             },
@@ -178,6 +186,8 @@ const GlobalStore = {
             commit('setFan8Info',  Fan8Info );
             commit('setFan9Info',  Fan9Info );
             commit('setCpldInfo',  CpldInfo );
+            commit('setsg1684xInfo', sg1684xInfo);
+            commit('setsg1684xTmpInfo', sg1684xTmpInfo);
             if (State === 'Quiesced' || State === 'InTest') {
               // OpenBMC's host state interface is mapped to 2 Redfish
               // properties "Status""State" and "PowerState". Look first
