diff --git a/src/store/plugins/WebSocketPlugin.js b/src/store/plugins/WebSocketPlugin.js
index b5492a4..e39e1e6 100644
--- a/src/store/plugins/WebSocketPlugin.js
+++ b/src/store/plugins/WebSocketPlugin.js
@@ -16,7 +16,8 @@ const WebSocketPlugin = (store) => {
       '/xyz/openbmc_project/gpio/fan',
       '/xyz/openbmc_project/gpio/hdd',
       '/xyz/openbmc_project/gpio/sata',
-      '/xyz/openbmc_project/gpio/identifyLed'
+      '/xyz/openbmc_project/gpio/identifyLed',
+      '/xyz/openbmc_project/sensors/temperature/AiCard_Temp',
     ],
     interfaces: [
       'xyz.openbmc_project.State.Host',
@@ -25,6 +26,7 @@ const WebSocketPlugin = (store) => {
       'xyz.openbmc_project.Gpio.Hdd',
       'xyz.openbmc_project.Gpio.Sata',
       'xyz.openbmc_project.Gpio.identifyLed',
+      'xyz.openbmc_project.Sensor.Value',
     ],
   };
 
@@ -55,7 +57,8 @@ const WebSocketPlugin = (store) => {
       } else if (/* (path === '/xyz/openbmc_project/gpio/fan') ||
                  (path === '/xyz/openbmc_project/gpio/hdd') ||
                  (path === '/xyz/openbmc_project/gpio/sata')|| */
-                 (path === '/xyz/openbmc_project/gpio/identifyLed')) {
+                 (path === '/xyz/openbmc_project/gpio/identifyLed') ||
+                 (path === '/xyz/openbmc_project/sensors/temperature/AiCard_Temp')) {
         store.dispatch('global/getSystemInfo');
       }
     };
