diff --git a/src/views/HardwareStatus/Sensors/Sensors.vue b/src/views/HardwareStatus/Sensors/Sensors.vue
index 6329d9d..4eb0cf2 100644
--- a/src/views/HardwareStatus/Sensors/Sensors.vue
+++ b/src/views/HardwareStatus/Sensors/Sensors.vue
@@ -1,22 +1,36 @@
 <template>
   <b-container fluid="xl">
     <page-title />
-    <b-row class="align-items-end">
-      <b-col sm="6" md="5" xl="4">
-        <search
-          :placeholder="$t('pageSensors.searchForSensors')"
-          data-test-id="sensors-input-searchForSensors"
-          @change-search="onChangeSearchInput"
-          @clear-search="onClearSearchInput"
-        />
+    <b-row class="align-items-end" style="display: flex;">
+      <b-col sm="2" md="3" xl="4">
+          <search
+            :placeholder="$t('pageSensors.searchForSensors')"
+            data-test-id="sensors-input-searchForSensors"
+            @change-search="onChangeSearchInput"
+            @clear-search="onClearSearchInput"
+          />
       </b-col>
-      <b-col sm="3" md="3" xl="2">
+      <b-col sm="1" md="2" xl="3">
         <table-cell-count
           :filtered-items-count="filteredRows"
           :total-number-of-cells="allSensors.length"
         ></table-cell-count>
       </b-col>
-      <b-col sm="3" md="4" xl="6" class="text-right">
+      <b-col sm="2" md="3" xl="4" v-if="sg1684xStatusIcon === 'success'">
+        <dl>
+          <b-row>
+            <b-col sm="1" md="2" xl="3">
+              <dt style="margin-right: 1px;">
+                {{ $t('pageSensors.1684X_Exist') }}  <status-icon :status="sg1684xStatusIcon" />
+              </dt>
+            </b-col>
+            <b-col>
+              <span style="margin-left: 1px;">{{ get1684xtemperature }} °C</span>
+            </b-col>
+          </b-row>
+        </dl>
+      </b-col>
+      <b-col sm="3" md="2" xl="1" class="text-right" >
         <table-filter :filters="tableFilters" @filter-change="onFilterChange" />
       </b-col>
     </b-row>
@@ -219,6 +233,23 @@ export default {
     filteredSensors() {
       return this.getFilteredTableData(this.allSensors, this.activeFilters);
     },
+    sg1684xStatus() {
+      return this.$store.getters['global/sg1684xStatus'];
+    },
+    sg1684xStatusIcon() {
+      switch (this.sg1684xStatus) {
+        case 'Exist':
+          return 'success';
+        default:
+          return 'secondary';
+      }
+    },
+    sg1684xTmp() {
+      return this.$store.getters['global/sg1684xTmp'];
+    },
+    get1684xtemperature() {
+      return this.sg1684xTmp; // 返回温度值
+    }
   },
   created() {
     this.startLoader();
