From 762783417ec1beca1769fcb93512aee32f78e915 Mon Sep 17 00:00:00 2001
From: Jamin Lin <jamin_lin@aspeedtech.com>
Date: Mon, 11 Jul 2022 14:18:31 +0800
Subject: [PATCH] Add to install libmctp-log.h

Signed-off-by: Jamin Lin <jamin_lin@aspeedtech.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5b69479..096528d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,4 +86,4 @@ target_link_libraries (test_asti3c mctp_intel)
 add_test (NAME asti3c COMMAND test_asti3c)
 
 install (TARGETS mctp_intel DESTINATION lib)
-install (FILES libmctp.h libmctp-cmds.h libmctp-vdpci.h libmctp-smbus.h libmctp-astpcie.h libmctp-msgtypes.h libmctp-asti3c.h DESTINATION include)
+install (FILES libmctp.h libmctp-cmds.h libmctp-vdpci.h libmctp-smbus.h libmctp-astpcie.h libmctp-msgtypes.h libmctp-asti3c.h libmctp-log.h DESTINATION include)
-- 
2.17.1

