From 00d39c43d01dbf06fbeb9cddc8a78e5a7bbe8bf1 Mon Sep 17 00:00:00 2001
From: Jamin Lin <jamin_lin@aspeedtech.com>
Date: Tue, 19 Jul 2022 17:26:33 +0800
Subject: [PATCH] disable all warnings being treated as errors

Signed-off-by: Jamin Lin <jamin_lin@aspeedtech.com>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 096528d..eaca8a1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,6 @@ project (libmctp)
 set(CMAKE_C_FLAGS
     "${CMAKE_C_FLAGS} \
     -Wall \
-    -Werror \
     -Wextra \
     -Wunused \
     -Wnull-dereference \
-- 
2.25.1

