diff --git a/src/HwmonTempMain.cpp b/src/HwmonTempMain.cpp
index 30d0075..adfca71 100644
--- a/src/HwmonTempMain.cpp
+++ b/src/HwmonTempMain.cpp
@@ -76,6 +76,7 @@ static const I2CDeviceTypeMap sensorTypes{
     {"TMP75", I2CDeviceType{"tmp75", true}},
     {"W83773G", I2CDeviceType{"w83773g", true}},
     {"TMP451", I2CDeviceType{"tmp451", true}},
+    {"CT7451", I2CDeviceType{"ct7451", true}},
 };
 
 static struct SensorParams
