From a966f851686731d73146842ce087377f2504d6f9 Mon Sep 17 00:00:00 2001
From: Troy Lee <troy_lee@aspeedtech.com>
Date: Thu, 6 May 2021 14:23:41 +0800
Subject: [PATCH 1/4] bmcweb: fixes virtual media buffer overflow

---
 include/vm_websocket.hpp | 23 ++++++++++++++++++++---
 1 file changed, 20 insertions(+), 3 deletions(-)

diff --git a/include/vm_websocket.hpp b/include/vm_websocket.hpp
index 5adf303..7c6f812 100644
--- a/include/vm_websocket.hpp
+++ b/include/vm_websocket.hpp
@@ -19,7 +19,7 @@ static crow::websocket::Connection* session = nullptr;
 // The max network block device buffer size is 128kb plus 16bytes
 // for the message header:
 // https://github.com/NetworkBlockDevice/nbd/blob/master/doc/proto.md#simple-reply-message
-static constexpr auto nbdBufferSize = 131088;
+static constexpr auto nbdBufferSize = (128*1024+16)*4;
 
 class Handler : public std::enable_shared_from_this<Handler>
 {
@@ -46,7 +46,16 @@ class Handler : public std::enable_shared_from_this<Handler>
         {
             return;
         }
-        proxy.wait();
+
+        std::error_code ec;
+        do
+        {
+            proxy.wait(ec);
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR << "Error on proxy.wait():" << ec;
+            }
+        } while (ec == std::errc::interrupted);
     }
 
     void connect()
@@ -204,7 +213,15 @@ inline void requestRoutes(App& app)
         })
         .onmessage([](crow::websocket::Connection& conn,
                       const std::string& data, bool) {
-            if (data.length() > handler->inputBuffer->capacity())
+            if (handler->inputBuffer->size() > 0)
+            {
+                BMCWEB_LOG_DEBUG << "Data length " << data.length()
+                                 << " iBuffer.cap() " << handler->inputBuffer->capacity()
+                                 << " iBuffer.size() " << handler->inputBuffer->size();
+            }
+
+            if (data.length() >
+                    handler->inputBuffer->capacity() - handler->inputBuffer->size())
             {
                 BMCWEB_LOG_ERROR << "Buffer overrun when writing "
                                  << data.length() << " bytes";
-- 
2.25.1

